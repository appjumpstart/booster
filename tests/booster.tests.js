const { join } = require('path')
const { test } = require('@ianwalter/bff')
const execa = require('execa')
const Knex = require('knex')
const config = require('./fixtures/knexfile')

const booster = '../../index.js'
const cwd = join(__dirname, 'fixtures')

test('default database migration and seeding', async ({ expect }) => {
  const knex = Knex(config.development)
  try {
    const { stdout } = await execa(booster, { cwd })
    expect(stdout).toMatchSnapshot()
    const { rows } = await knex.raw('SELECT * FROM categories')
    expect(rows[0].name).toBe('Personal')
    expect(rows[1].name).toBe('Business')
  } finally {
    await knex.destroy()
  }
})

test('specified database migration and seeding', async ({ expect }) => {
  const knex = Knex(config.test)
  try {
    const { stdout } = await execa(booster, ['test'], { cwd })
    expect(stdout).toMatchSnapshot()
    const { rows } = await knex.raw('SELECT * FROM categories')
    expect(rows[0].name).toBe('Personal')
    expect(rows[1].name).toBe('Business')
  } finally {
    await knex.destroy()
  }
})

test('migration and seeding using specified knexfile', async ({ expect }) => {
  const knex = Knex(config.development)
  try {
    const { stdout } = await execa(booster, ['-c', cwd], { cwd })
    expect(stdout).toMatchSnapshot()
    const { rows } = await knex.raw('SELECT * FROM categories')
    expect(rows[0].name).toBe('Personal')
    expect(rows[1].name).toBe('Business')
  } finally {
    await knex.destroy()
  }
})

test('error thrown when specified knexfile not found', async ({ expect }) => {
  const cmd = execa(booster, ['-c', '/tmp'], { cwd })
  await expect(cmd).rejects.toThrowErrorMatchingSnapshot()
})
