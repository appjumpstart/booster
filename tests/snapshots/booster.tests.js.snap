// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default database migration and seeding 1`] = `
"ğŸš€  [1mDatabase booster created![22m
ğŸ‘Ÿ  [1mMigrations run![22m
ğŸŒ±  [1mDatabase seeded![22m"
`;

exports[`error thrown when specified knexfile not found 1`] = `"Command failed with exit code 1 (EPERM): ../../index.js -c /tmp"`;

exports[`migration and seeding using specified knexfile 1`] = `
"ğŸ‘Ÿ  [1mMigrations run![22m
ğŸŒ±  [1mDatabase seeded![22m"
`;

exports[`specified database migration and seeding 1`] = `
"ğŸš€  [1mDatabase booster_test created![22m
ğŸ‘Ÿ  [1mMigrations run![22m
ğŸŒ±  [1mDatabase seeded![22m"
`;
